Dima/netflix-roulette/src/Components/Search/Search.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import Button from '../../Shared/Buttons/Button'
+import { parse } from 'query-string'
+import * as H from 'history'
+
+interface ISearchProps{
+  handleSearchChange({ search, searchBy }: {search: string, searchBy:string}): void,
+  location: H.Location
+};
+
+enum IsearchParams {
+  Title = 'title',
+  Genre = 'genre'
+}
+
+export default class Search extends React.Component<ISearchProps> {
+
+  state = {
+    searchInput: '',
+    searchBy: 'title',
+  };
+
+  componentDidMount() {
+    const URLData = parse(this.props.location.search) as {
+      searchBy: string;
+      search: string
+    };
+    let { searchBy, search } = URLData;
+    if(searchBy) {
+      searchBy === IsearchParams.Title
+        ? this.setState({ searchBy: IsearchParams.Title }) 
+          : this.setState({ searchBy: IsearchParams.Genre })
+    } else {
+      this.setState({ searchBy: IsearchParams.Title })
+      search ? this.setState({ searchInput: search }) : search = ''
+    };
+  };
```

Comment:
search = '' ??
You have an issue here. What if there will be both searchBy and search ??